src/core/server/integration_tests/elasticsearch/version_compatibility.test.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  createTestServers,
+  type TestElasticsearchUtils,
+  type TestKibanaUtils,
+} from '@kbn/core-test-helpers-kbn-server';
+import { esTestConfig } from '@kbn/test';
+import { firstValueFrom, Subject } from 'rxjs';
+import { CliArgs } from '@kbn/config';
+import Semver from 'semver';
+
+function nextMinor() {
+  return Semver.inc(esTestConfig.getVersion(), 'minor') || '10.0.0';
+}
+
+describe('Version Compatibility', () => {
+  let esServer: TestElasticsearchUtils | undefined;
+  let kibanaServer: TestKibanaUtils | undefined;
+  let abortController: AbortController | undefined;
+  let consoleSpy: jest.SpyInstance;
+
+  beforeEach(() => {
+    consoleSpy = jest.spyOn(console, 'log');
+  });
+
+  afterEach(async () => {
+    consoleSpy.mockRestore();
+    if (kibanaServer) {
+      await kibanaServer.stop();
+    } else {
+      abortController?.abort();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+    kibanaServer = undefined;
+    abortController = undefined;
+    esServer = undefined;
+  });
+
+  const startServers = async ({
+    cliArgs,
+    customKibanaVersion,
+    ignoreVersionMismatch,
+  }: {
+    cliArgs?: Partial<CliArgs>;
+    customKibanaVersion?: string;
+    ignoreVersionMismatch?: boolean;
+  } = {}) => {
+    const { startES, startKibana } = createTestServers({
+      adjustTimeout: jest.setTimeout,
+      settings: {
+        kbn: {
+          cliArgs,
+          customKibanaVersion,
+          elasticsearch: {
+            ignoreVersionMismatch,
+          },
+        },
+      },
+    });
+
+    esServer = await startES();
+    abortController = new AbortController();
+    kibanaServer = await startKibana(abortController.signal);
+  };
+
+  it('should start when versions match', async () => {
+    await expect(startServers({})).resolves.toBeUndefined();
+  });
+
+  it('should flag the incompatibility on version mismatch', async () => {
+    const found$ = new Subject<void>();
+    consoleSpy.mockImplementation((str) => {
+      if (str.includes('is incompatible')) {
+        found$.next();
+      }
+    });
+    await Promise.race([
+      firstValueFrom(found$),
+      startServers({ customKibanaVersion: nextMinor() }).then(() => {
```

Comment:
This test only asserts against version mismatch between x.n and x.n+1. We should test against a mismatch between x.n and x.n-1 too.

Can we assume mismatches between current and previous/next patches won't matter?

Reply:
> This test only asserts against version mismatch between x.n and x.n+1. We should test against a mismatch between x.n and x.n-1 too

Kibana "is compatible" with newer minor versions of ES already:

https://github.com/elastic/kibana/blob/35083504464a4950d5ce5c6464cab387dd98d0d8/packages/core/elasticsearch/core-elasticsearch-server-internal/src/version_check/es_kibana_version_compatability.ts#L29-L39

(which is unitary tested in the associated test file), so I'd say it's fine to only test the other scenario?

Reply:
I guess it's good to have an integration test for the "ES is next minor" use case to catch any potential regressions in the future.

I added it in [`d4e619a` (#156837)](https://github.com/elastic/kibana/pull/156837/commits/d4e619a76a7578dea39e3c096af9e1427faf7b0a)

Thank you for raising it, @TinaHeiligers